sdk/communication/Azure.Communication.CallAutomation/src/CallMedia.cs

```diff
@@ -428,8 +529,212 @@ private static PlaySourceInternal TranslatePlaySourceToInternal(PlaySource playS
                 sourceInternal.PlaySourceCacheId = fileSource.PlaySourceCacheId;
                 return sourceInternal;
             }
+            else if (playSource != null && playSource is TextSource textSource)
+            {
+                sourceInternal = new PlaySourceInternal(PlaySourceTypeInternal.Text);
+                sourceInternal.TextSource = new TextSourceInternal(textSource.Text);
+                sourceInternal.TextSource.SourceLocale = textSource.SourceLocale ?? null;
+                sourceInternal.TextSource.VoiceGender = textSource.VoiceGender ?? GenderType.Male;
+                sourceInternal.TextSource.VoiceName = textSource.VoiceName ?? null;
+                sourceInternal.TextSource.CustomVoiceEndpointId = textSource.CustomVoiceEndpointId ?? null;
+                sourceInternal.PlaySourceCacheId = textSource.PlaySourceCacheId;
+                return sourceInternal;
+            }
+            else if (playSource != null && playSource is SsmlSource ssmlSource)
+            {
+                sourceInternal = new PlaySourceInternal(PlaySourceTypeInternal.Ssml);
+                sourceInternal.SsmlSource = new SsmlSourceInternal(ssmlSource.SsmlText);
+                sourceInternal.SsmlSource.CustomVoiceEndpointId = ssmlSource.CustomVoiceEndpointId ?? null;
+                sourceInternal.PlaySourceCacheId = ssmlSource.PlaySourceCacheId;
+                return sourceInternal;
+            }
             else
             { return null; }
         }
+
+        /// <summary>
+        /// Starts continuous Dtmf recognition.
+        /// </summary>
+        /// <param name="targetParticipant">A target participant identifier for starting continuous Dtmf recognition.</param>
+        /// <param name="operationContext">An optional context object containing information about the operation, such as a unique identifier or custom metadata.</param>
+        /// <param name="cancellationToken">An optional CancellationToken to cancel the request.</param>
+        /// <returns>Returns an HTTP response with a 200 status code for success, or an HTTP failure error code in case of an error.</returns>
+        public virtual Response StartContinuousDtmfRecognition(CommunicationIdentifier targetParticipant, string operationContext = default,
```

Comment:
I am guessing we have no plan to add option pattern here?

Reply:
yes, correct